jetty-http/src/main/java/org/eclipse/jetty/http/pathmap/UriTemplatePathSpec.java

```diff
@@ -321,7 +322,8 @@ private boolean isValidBasicLiteralCodepoint(int codepoint)
             int groupCount = matcher.groupCount();
             for (int i = 1; i <= groupCount; i++)
             {
-                ret.put(this.variables[i - 1],matcher.group(i));
+                String value = UrlEncoded.decodeString(matcher.group(i));
```

Comment:
Doesn't `UrlEncoded.decodeString()` require a charset?

Reply:
It uses the default (UTF-8, but configurable) in UrlEncoded.